core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -1218,9 +1219,9 @@ public Object clone() throws CloneNotSupportedException {
    * Throws an exception if the key is too large.
    *
    */
-  public void sanityCheckKey() {
-    // If fails the check
-    if (((long) this.row.length + (long) this.colFamily.length + (long) this.colQualifier.length + (long) this.colVisibility.length + (4L * 5L + 8L)) >= Integer.MAX_VALUE) {
+  private void sanityCheckKey() {
+	  //If the key is too large, we throw an exception. We subtract 5L from KEY_VALUE_OVERHEAD to remove the accounting for value overhead
+    if (((long) this.row.length + (long) this.colFamily.length + (long) this.colQualifier.length + (long) this.colVisibility.length + (RFile.KEY_VALUE_OVERHEAD-5L)) >= Integer.MAX_VALUE) {
```

Comment:
could have two constants in RFile : `KEY_OVERHEAD` and `VALUE_OVERHEAD` then the subtraction of `5L` is not needed here, just use `KEY_OVERHEAD`